src/main/java/com/razorpay/PaymentClient.java

```diff
@@ -31,15 +34,27 @@ public Refund refund(String id) throws RazorpayException {
   }
 
   public Refund refund(String id, JSONObject request) throws RazorpayException {
-    return post(String.format(Constants.PAYMENT_REFUND, id), request);
+    RefundClient refundClient = new RefundClient(auth);
```

Comment:
We don't want to create a new object every time refund is called. Move this to the PaymentClient constructor